src/components/CardForm/CardForm.tsx

```diff
@@ -0,0 +1,71 @@
+import { FormEvent } from 'react';
+import CardNumberSection from '../../components/CardNumberSection/CardNumberSection';
+import CardExpirationSection from '../../components/CardExpirationSection/CardExpirationSection';
+import CardCompanySection from '../../components/CardCompanySection/CardCompanySection';
+import CvcSection from '../../components/CvcSection/CvcSection';
+import PasswordSection from '../../components/PasswordSection/PasswordSection';
+import Button from '../../components/Button/Button';
+import { useNavigate } from 'react-router';
+import { useStack } from '../../hooks/useStack';
+import { useStepFlow } from '../../hooks/useStepFlow';
+import { STEPS } from '../../constants';
+import styles from '../../../src/pages/HomePage/HomePage.module.css';
+import useValues from '../../hooks/useValues';
+import { useFormValid } from '../../hooks/useFormValid';
+
+type CardFormProps = ReturnType<typeof useValues>['cardForm'];
+
+export function CardForm(props: CardFormProps) {
+  const { Stack, setStep } = useStack<(typeof STEPS)[number]>('카드번호');
+
+  const { cardNumbers, company, expiration, cvc, password } = props;
+
+  useStepFlow({ setStep, cvc: cvc.cvc, expiration: expiration.expiration, company: company.company, cardNumbers: cardNumbers.cardNumbers });
+
+  const navigate = useNavigate();
+
+  const handleGoCompletePage = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    navigate('/complete', {
+      state: {
+        firstCardNumber: cardNumbers.cardNumbers.first.value,
+        company: company.company
+      }
+    });
+  };
+
+  const buttonVisible = useFormValid({
```

Comment:
```suggestion
  const isSubmit = useFormValid({
```
`buttonVisible`도 맞는 표현이지만 폼을 제출할 수 있는지 여부를 나타내는 `isSubmit` 같은 네이밍도 좋은 것 같아요🙃